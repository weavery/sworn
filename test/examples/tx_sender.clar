(define-data-var sender (optional principal) none)

(define-read-only (read)
  (ok (var-get sender)))

(define-public (update)
  (begin
    (var-set sender (some tx-sender))
    (ok (var-get sender))))

(define-public (clear)
  (begin
    (var-set sender none)
    (ok (var-get sender))))
