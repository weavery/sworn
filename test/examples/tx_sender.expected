// Generated by Sworn

clarity.requireVersion("0.0")

function read(state) {
  return clarity.ok(state.sender);
}

function update(state) {
  state.sender = clarity.some(clarity.txSender());
  return {state, result: clarity.ok(state.sender)};
}

function clear(state) {
  state.sender = clarity.none;
  return {state, result: clarity.ok(state.sender)};
}

export function handle(state, action) {
  const input = action.input;
  if (input.function === 'read') {
    return {result: read(state)};
  }
  if (input.function === 'update') {
    return update(state);
  }
  if (input.function === 'clear') {
    return clear(state);
  }
  return {state};
}
